crypto/fipsmodule/rand/rand.c

```diff
@@ -312,6 +314,48 @@ static void rand_get_seed(struct rand_thread_state *state,
 
 #endif
 
+// CRYPTO_snapsafe_must_defend indicates whether snapsafe-type ube's must be
+// defended against.
+// Experimental support, so don't enforce anything yet.
+OPENSSL_INLINE int CRYPTO_snapsafe_must_defend(void) {
+#if defined(AWSLC_SNAPSAFE_MUST_DEFEND)
+  return 1;
+#else
+  return 0;
+#endif
+}
+
+// CRYPTO_fork_must_defend indicates whether fork-type ube's must be defended
+// against.
+OPENSSL_INLINE int CRYPTO_fork_must_defend(void) {
+  if (rand_fork_unsafe_buffering_enabled() == 0) {
+    return 1;
+  }
+  else {
+    return 0;
+  }
+}
+
+// must_defend_against_ube implements the logic that decides whether
+// we should enforce entropy mixing to safeguards against uniqeness breaking
+// events (ube). Currently, there are two mechanisms implemented that attempts
+// to detect such events. They are:
+//  * fork detection: attempts to detect process forks.
+//  * snapsafe detection: attempts to detect snapshot/VM resumes.
+// If a mechanism is not enabled, we check if something else has indicated that
+// the particular type of ube must be defended against.
+// Returns 1 if must defend and 0 otherwise.
```

Comment:
Mere suggestion :)
```suggestion
// If a mechanism is not enabled, but a defense against ube was still enforced 
// (as indicated by the functions above), return 1; otherwise, return 0.
```